engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnDefinitionsServiceImpl.java

```diff
@@ -419,43 +460,147 @@ public Dampening addGroupDampening(String tenantId, Dampening groupDampening) th
 
     @Override
     public void removeDampening(String tenantId, String dampeningId) throws Exception {
+        if (isEmpty(tenantId)) {
+            throw new IllegalArgumentException("TenantId must be not null");
+        }
+        if (isEmpty(dampeningId)) {
+            throw new IllegalArgumentException("dampeningId must be not null");
+        }
 
+        Dampening dampening = getDampening(tenantId, dampeningId);
+        if (null == dampening) {
+            log.debugf("Ignoring removeDampening(%s), the Dampening does not exist.", dampeningId);
+            return;
+        }
+
+        String triggerId = dampening.getTriggerId();
+        Trigger trigger = getTrigger(tenantId, triggerId);
+
+        if (trigger.isGroup()) {
+            throw new IllegalArgumentException("Trigger [" + tenantId + "/" + triggerId + "] is a group trigger.");
+        }
+        if (trigger.isMember() && !trigger.isOrphan()) {
+            throw new IllegalArgumentException("Trigger [" + tenantId + "/" + triggerId
+                    + "] is a member trigger and must be managed via the group.");
+        }
+
+        removeDampening(dampening);
     }
 
     @Override
     public void removeGroupDampening(String tenantId, String groupDampeningId) throws Exception {
 
     }
 
+    private void removeDampening(Dampening dampening) throws Exception {
+        try {
+            backend.remove(pk(dampening));
+        } catch (Exception e) {
+            log.errorDatabaseException(e.getMessage());
+            throw e;
+        }
+
+        if (null != alertsEngine) {
+            alertsEngine.reloadTrigger(dampening.getTenantId(), dampening.getTriggerId());
+        }
+
+        notifyListeners(new DefinitionsEvent(Type.DAMPENING_CHANGE, dampening));
+    }
+
     @Override
     public Dampening updateDampening(String tenantId, Dampening dampening) throws Exception {
-        return null;
+        if (isEmpty(tenantId)) {
+            throw new IllegalArgumentException("TenantId must be not null");
+        }
+        if (isEmpty(dampening)) {
+            throw new IllegalArgumentException("Dampening, DampeningId and TriggerId must be not null");
+        }
+
+        checkTenantId(tenantId, dampening);
+
+        String triggerId = dampening.getTriggerId();
+        Trigger trigger = getTrigger(tenantId, triggerId);
+
+        if (trigger.isGroup()) {
+            throw new IllegalArgumentException("Trigger [" + tenantId + "/" + triggerId + "] is a group trigger.");
+        }
+        if (trigger.isMember() && !trigger.isOrphan()) {
+            throw new IllegalArgumentException("Trigger [" + tenantId + "/" + triggerId
+                    + "] is a member trigger and must be managed via the group.");
+        }
+
+        return updateDampening(dampening);
+    }
+
+    private Dampening updateDampening(Dampening dampening) throws Exception {
+        try {
+            backend.put(pk(dampening), dampening);
```

Comment:
Ha, good eyes.  Of course the bug was in the method for which I forgot to add a test.  Fixing...